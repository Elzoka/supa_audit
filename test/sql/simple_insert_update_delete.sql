create table public.members(
    id int primary key,
    name text not null
);

create trigger t
    before insert or update or delete
    on public.members
    for each row
    execute procedure audit.insert_update_delete_trigger();

insert into public.members(id, name)
values (1, 'foo');

update public.members
    set name = 'bar'
    where id = 1;

delete from public.members;

select
    id,
    record_id,
    old_record_id,
    op,
    table_oid,
    table_schema,
    table_name,
    record
from
    audit.record_version;
