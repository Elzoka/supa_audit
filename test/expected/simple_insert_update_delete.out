begin;
    create table public.members(
        id int primary key,
        name text not null
    );
    select audit.enable_tracking('public.members');
 enable_tracking 
-----------------
 
(1 row)

    insert into public.members(id, name)
    values (1, 'foo');
    update public.members
        set name = 'bar'
        where id = 1;
    delete from public.members;
    select
        id,
        record_id,
        old_record_id,
        op,
        table_schema,
        table_name,
        record
    from
        audit.record_version;
 id |              record_id               |            old_record_id             |   op   | table_schema | table_name |          record          
----+--------------------------------------+--------------------------------------+--------+--------------+------------+--------------------------
  2 | 13979db9-fa66-54eb-87eb-10ff957d6a63 |                                      | INSERT | public       | members    | {"id": 1, "name": "foo"}
  3 | 13979db9-fa66-54eb-87eb-10ff957d6a63 | 13979db9-fa66-54eb-87eb-10ff957d6a63 | UPDATE | public       | members    | {"id": 1, "name": "bar"}
  4 |                                      | 13979db9-fa66-54eb-87eb-10ff957d6a63 | DELETE | public       | members    | 
(3 rows)

rollback;
